platform/platform-api/src/main/java/org/codice/ddf/platform/api/Describable.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.api;
+
+/**
+ * Describable is used to capture a basic description.
+ */
+public interface Describable {
+    /**
+     * Retrieve the version.
+     *
+     * @return the version of the item being described (example: 1.0)
+     */
+    String getVersion();
+
+    /**
+     * Returns the name, aka ID, of the describable item. The name should be unique for each
```

Comment:
May want to relax the uniqueness constraint a little here. It should be unique within the scope of a service or component, i.e., unique within the scope of `Migratable`s, but not between `Migratable`s and Catalog's metacards.